LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -132,6 +144,23 @@ public final class DoseStore {
     }
     private let lockedInsulinSensitivitySchedule: Locked<InsulinSensitivitySchedule?>
 
+    /// The insulin sensitivity schedule, applying recent overrides relative to the current moment in time.
+    public var insulinSensitivityScheduleApplyingOverrideHistory: InsulinSensitivitySchedule? {
+        if let insulinSensitivitySchedule = insulinSensitivitySchedule {
+            return overrideHistory?.resolvingRecentInsulinSensitivitySchedule(insulinSensitivitySchedule)
+        } else {
+            return nil
+        }
+    }
+
+    /// The computed EGP schedule based on the basal profile and insulin sensitivity schedule.
+    public var egpSchedule: EGPSchedule? {
```

Comment:
That's correct, and similarly for the carb sensitivity schedule. These schedules are more reflective of the actual physiology that basals and carb ratios intend to model, so they serve as a valuable diagnostic for poor settings.
In the future, when we get to tuning user parameters algorithmically, it might be worthwhile (if not necessary) to switch to using these schedules as the underlying model (at which point the basal and carb ratio schedules become the derived ones).

Reply:
💯% agree!